capstone/src/main/webapp/script.js

```diff
@@ -12,48 +12,78 @@
+          addRestaurant(restaurant, restaurantsList);
+        });
+      });
     }
-  }
+  });
 }
 
-/**
- * Creates an element for a restaurant, including
- * redirecting to its detail page on click.
- */
-function createRestaurantElement(restaurant) {
-  // Article tag to encapsulate restaurant elements
-  const restaurantElement = document.createElement('h2');
-  const linkElement = document.createElement('a');
+function addRestaurant(restaurant, containerElement) {
+  const restaurantDiv = document.createElement('div');
+  restaurantDiv.classList.add('mdc-card', 'restaurant-container');
+
+  const restaurantNameDiv = createRestaurantNameDiv(restaurant);
+  const restaurantCuisineDiv = createRestaurantCuisineDiv(restaurant);
+  const restaurantStrugglingDiv = createRestaurantIsStrugglingDiv(restaurant);
 
-  // Restaurant element UI details
-  const redirect =
-      '/restaurantDetails.html?restaurantKey=' + restaurant.restaurantKey;
-  linkElement.setAttribute('href', redirect);
+  restaurantDiv.appendChild(restaurantNameDiv);
+  restaurantDiv.appendChild(restaurantCuisineDiv);
+  restaurantDiv.appendChild(restaurantStrugglingDiv);
+
+  containerElement.appendChild(restaurantDiv);
+}
+
+function createRestaurantNameDiv(restaurant) {
+  const params = new URLSearchParams();
+  params.append('restaurantKey', restaurant.restaurantKey);
+
+  const linkElement = document.createElement('a');
+  linkElement.classList.add('restaurant-name-link');
   linkElement.innerText = restaurant.name;
-  restaurantElement.appendChild(linkElement);
+  linkElement.href = '/restaurantDetails?' + params;
+
+  const restaurantNameHeader = document.createElement('h3');
+  restaurantNameHeader.appendChild(linkElement);
+
+  const restaurantNameDiv = document.createElement('div');
+  restaurantNameDiv.classList.add(
+      'restaurant-name-container', 'mdc-typography--headline6');
+  restaurantNameDiv.appendChild(restaurantNameHeader);
+
+  return restaurantNameDiv;
+}
+
+function createRestaurantCuisineDiv(restaurant) {
+  const restaurantCuisineDiv = document.createElement('div');
+  restaurantCuisineDiv.classList.add('restaurant-cuisine-container');
+  restaurant.cuisine.forEach((item) => {
+    const cuisineItemButton = document.createElement('button');
+    cuisineItemButton.innerHTML = item;
+    cuisineItemButton.classList.add('mdc-button', 'mdc-ripple-surface');
+    restaurantCuisineDiv.appendChild(cuisineItemButton);
+  });
+
+  return restaurantCuisineDiv;
+}
+
+function createRestaurantIsStrugglingDiv(restaurant) {
+  const restaurantStrugglingDiv = document.createElement('div');
+  restaurantStrugglingDiv.classList.add('restaurant-struggling-container');
+  if (restaurant.isStruggling) {
+    restaurantStrugglingDiv.innerHTML = '<p>This restaurant needs help!</p>';
+  } else {
+    restaurantStrugglingDiv.innerHTML = '<p>Keep this restaurant growing!</p>';
+  }
 
-  return restaurantElement;
+  return restaurantStrugglingDiv;
 }
```

Comment:
At some point, I think it'd be great to add method documentation before all the methods in this file.

Reply:
Okay, I can work on that now. Sorry about that.

Reply:
Done.